app/pages/patient/DataDonationConsentDialog.js

```diff
@@ -0,0 +1,362 @@
+  const [currentConsentStep, setCurrentConsentStep] = React.useState(0);
+  const [scrolledToBottom, setScrolledToBottom] = React.useState(false);
+  const [caregiverName, setCaregiverName] = React.useState(caregiverNameProp);
+  const { [DATA_DONATION_CONSENT_TYPE]: consentDocument } = useSelector((state) => state.blip.consents);
+
+  const nameSchema = yup.object().shape({
+    name: yup.string().required(t('Parent or Legal Guardian Name is required')),
+  });
+
+  const createConsentSchema = prefix => yup.object().shape({
+    [`${prefix}ConsentRead`]: yup.boolean().oneOf([true], t('You must read the consent statement before proceeding')),
+    [`${prefix}Consent`]: yup.boolean().oneOf([true], t('You must agree to consent before proceeding')),
+  });
+
+  const primaryConsentSchema = createConsentSchema('primary');
+  const secondaryConsentSchema = createConsentSchema('secondary');
+
+  const schemas = {
+    primary: patientAssentRequired && accountType === 'personal' ? nameSchema.concat(primaryConsentSchema) : primaryConsentSchema,
+    secondary: secondaryConsentSchema,
+  };
+
+  const formikContext = useFormik({
+    initialValues: {
+      name: caregiverName,
+      primaryConsentRead: false,
+      primaryConsent: false,
+      secondaryConsentRead: false,
+      secondaryConsent: false,
+    },
+    validationSchema: schemas[formSteps[currentConsentStep]],
+    onSubmit: (values, formikHelpers) => {
+      const nextStep = currentConsentStep + 1;
+      if (formSteps[nextStep]) {
+        setCurrentConsentStep(nextStep);
+        setCaregiverName(values.name);
+        formikHelpers.resetForm();
+        document.getElementById('consentDocumentText')?.scrollTo({ top: 0 });
+        setScrolledToBottom(false);
+      } else {
+        onConfirm({ ...values, name: caregiverName });
+      }
+    },
+  });
+
+  useEffect(() => {
+    if (isString(formikContext.values.name)) setCaregiverName(formikContext.values.name);
+  }, [formikContext.values.name]);
+
+  const consentText = getConsentText(accountType, patientAgeGroup, patientName, caregiverName, consentDate);
+  const consentQuestion = consentText[`${formSteps[currentConsentStep]}ConsentQuestion`];
+  const consentDescription = consentText[`${formSteps[currentConsentStep]}ConsentDescription`];
+  const consentInputLabel = consentText[`${formSteps[currentConsentStep]}ConsentInputLabel`];
+  const consentReviewMessage = consentText[`${formSteps[currentConsentStep]}ConsentReviewMessage`];
+  const consentSignature = consentText[`${formSteps[currentConsentStep]}ConsentSignature`];
+  const consentNameInputLabel = consentText[`${formSteps[currentConsentStep]}ConsentNameInputLabel`];
+
+  const handleConsentDocumentScroll = (e) => {
+    const { scrollTop, scrollHeight, clientHeight } = e.target;
+    // Use a small threshold (5px) to account for rendering differences
+    if (scrollTop + clientHeight >= scrollHeight - 5) {
```

Comment:
User must scroll through consent document before they can check the consent checkbox.